pkg/remote/remote.go

```diff
@@ -91,22 +81,27 @@ func WithDetection() heartbeat.HandleOption {
 				if err != nil {
 					log.Errorf("failed to create new remote client: %s", err)
 
+					deleteLocalFile(tmpFile.Name())
+
 					continue
 				}
 
 				err = c.DownloadFile(tmpFile.Name())
 				if err != nil {
 					log.Errorf("failed to download file to temporary folder: %s", err)
 
+					deleteLocalFile(tmpFile.Name())
+
 					continue
 				}
 
-				hh[i].LocalFile = tmpFile.Name()
-				// we save untouched entity for offline handling
-				hh[i].EntityRaw = h.Entity
+				h.LocalFile = tmpFile.Name()
+				h.EntityRaw = h.Entity // save untouched entity for offline handling
```

Comment:
In Go the coments are usually put before the declaration or the block code as we had before this line changes.